### PR TITLE
WIP: check task before cancellation

### PR DESCRIPTION
**Pull request by MaChengxin**
Rebase the commit on `next` instead of `devel`

### CODE REVIEW

**Comment by devos50** [comment]
@MaChengxin please cover the missing lines with a basic unit tests (see the following report: https://jenkins.tribler.org/job/GH_Tribler_PR_coverage_check/4883//artifact/coverage_report.html). For an example, see the https://github.com/Tribler/tribler/blob/next/Tribler/Test/Core/TorrentChecker/test_torrentchecker_session.py file. You should add a new test there or modify existing tests so the lines are covered.

@qstokkink if by chance you are at EWI, could you restart the Jenkins slave on the Mac? The clocks seems to be out of sync.

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
@devos50 MAC should be live again

**Comment by devos50** [comment]
@qstokkink the clock difference still seems to be 2:30 minutes. Could you check whether the system time is right on the mac machine?

**Comment by qstokkink** [comment]
@devos50 figured out how to correct the time 👍

**Comment by qstokkink** [comment]
retest this please

**Comment by devos50** [comment]
Seems to work again, nice!

File: Tribler/Test/Core/TorrentChecker/test_torrentchecker_session.py [hunk 1]
```diff
@@ -275,6 +275,17 @@ def on_error(failure):
 
         return test_deferred
 
+    import mock
+
+    def dummy_is_pending_task_active(*args, **kwargs):
+        return True
+
+    @mock.patch("Tribler.Core.TorrentChecker.session.UdpTrackerSession.is_pending_task_active",
```
Comment (devos50) [inline Tribler/Test/Core/TorrentChecker/test_torrentchecker_session.py]: Don't use the `mock` framework; take a look at the other tests in the same file to see how you can do it without the specified library.

Comment (MaChengxin) [inline Tribler/Test/Core/TorrentChecker/test_torrentchecker_session.py]: I will do that. Actually this isn't working and I just wanted to save it at the end of the day.
(I didn't know that it was updated here automatically after I pushed it to my own repository.)

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by devos50** [comment]
I'll close this PR since it probably won't be merged anytime soon.

**Pull request closed without merging**